### DISCUSSION TITLE
Modifying existing objects through the HFSS API.

### DISCUSSION THREAD

**Discussion by yxjhlxw** [ansys/pyaedt#3508] [Q&A] [answered]
Hello everyone, I'm new to pyaedt and I'm currently trying to use the HFSS API for my model optimization work. I would like to know how I can change the position and size of my existing objects through code. I would greatly appreciate any guidance you can provide. Thank you in advance.

**Comment by Samuelopez-ansys** [comment, marked as answer]
@yxjhlxw , you can do it through the object history. For instance, if you have a Box called "Box1":

```
q = pyaedt.Hfss(specified_version="2023.1")

q.modeler["Box1"].history().props["XSize"] = "5mm"

q.modeler["Box1"].move([5, 5, 5])

q.release_desktop(False, False)
```

For the position, you can modify the properties or directly move the object. My suggestion is to create a parameter and control the size and position, but it is up to you.

Regards

**Reply by yxjhlxw** [reply]
Thank you very much for your answer and assistance!
